Add first version of package browsing experience

balajikris opened the pull request:
Adds a new command that lists all azure-sdk-for-node packages from npm
and user can filter the list and select a particular package. Hitting
Enter after selection adds a dependency to the selected package in the
project's package.json and adds a require statement in the active
document, if one is open.

Addresses https://github.com/Azure/azure-node-essentials/issues/8

balajikris committed:
Add first version of package browsing experience

Adds a new command that lists all azure-sdk-for-node packages from npm
and user can filter the list and select a particular package. Hitting
Enter after selection adds a dependency to the selected package in the
project's package.json and adds a require statement in the active
document, if one is open.

msftclas commented:
@balajikris,
Thanks for your contribution as a Microsoft full-time employee or intern. You do not need to sign a CLA.
_Thanks,
Microsoft Pull Request Bot_

balajikris committed:
add package-browsing gif illustrating how to use

the feature.

balajikris committed:
update READEME with package-browsing feature help

balajikris commented:
/cc: @amarzavery

balajikris commented:
tagging @fiveisprime and @lostintangent . I thought you might be interested in this, as it is kinda related to your North star goal for "Node.js Development with Visual Studio Code and Azure". I have this extension for VSCode which is an accompaniment to Node-SDK-for-Azure (feel free to check out what it does in the readme). 

This particular PR, **adds the ability to browse, select and install Node-SDK packages** without leaving the code editor. [See a screenshot here](https://raw.githubusercontent.com/balajikris/azure-node-essentials-1/cdf852c9977993a940a635a5cf0b25f0d2659a78/resources/help/package-browsing.gif).

I'd love to hear your thoughts or feedback with regards to this feature or the project itself in general.

balajikris commented on src/commands/azure-node.browse-packages.js:
```diff
@@ -0,0 +1,60 @@
+
+exports.createCommand = function createCommand() {
+  vscode.commands.registerCommand('Azure-Node.browse-packages', function () {
+    npmUserPackages('windowsazure').then(data => {
```
ToDo: Move this to extension load from command invocation. Two advantages:
1. perf will be better if we didn't make a webrequest after command invocation.
2. prefetching and caching will also ensure we don't make a request on every command invocation but once per VSCode session.

balajikris committed:
small conditional check and formatting.

balajikris committed:
update latest generator package version

balajikris committed:
run npm install after updating package.json

balajikris committed:
safety check

balajikris committed:
updates to changelog

balajikris committed:
slight improvements to generated code

for template deployment.

balajikris committed:
Fix npm install after adding entry to..

.. package.json. Without the proper prefix, npm will install the package
to the wrong location and will fail. pass prefix to ask it to install in
the current project's node_modules folder.
